Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -542,6 +553,46 @@ QualifiedName MarkupAccumulator::xmlAttributeSerialization(const Attribute& attr
     return prefixedName;
 }
 
+LocalFrame* MarkupAccumulator::frameForAttributeReplacement(const Element& element)
+{
+    if (inXMLFragmentSerialization() || m_replacementURLStrings.isEmpty())
+        return nullptr;
+
+    auto* currentElement = const_cast<Element*>(&element);
+    if (!is<HTMLIFrameElement>(currentElement))
+        return nullptr;
+
+    auto& iframeElement = downcast<HTMLIFrameElement>(*currentElement);
+    return dynamicDowncast<LocalFrame>(iframeElement.contentFrame());
+}
+
+Attribute MarkupAccumulator::replaceAttributeIfNecessary(const Element& element, const Attribute& attribute)
+{
+    if (!element.isHTMLContentAttribute(attribute))
+        return attribute;
+
+    auto frame = frameForAttributeReplacement(element);
+    if (!frame || !frame->loader().documentLoader()->response().url().isAboutSrcDoc())
```

Comment:
How about data URL? It seems like we can skip this step as well.

Reply:
Thanks for bringing this up. Data URL and blob URL are on my to-do list and I am going to look at them next.